AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -281,14 +281,17 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
                 dispatchedModelSynced.set(false)
             }
             if storageEngine == nil {
-
-                completion(.successfulVoid)
+                workQueue.async {
+                    completion(.successfulVoid)
+                }
                 return
             }
 
             storageEngine.stopSync { result in
-                self.storageEngine = nil
-                completion(result)
+                self.workQueue.async {
+                    self.storageEngine = nil
```

Comment:
we should set this to `nil` outside of the async block, because the sync block performs work serially, so that access to `storageEngine` is serial, thus putting it inside the async block will defeat that purpose